From fad2197bcb570350cb03c4ed789015baf0f86c81 Mon Sep 17 00:00:00 2001
From: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
Date: Thu, 16 Sep 2010 16:30:12 -0700
Subject: [PATCH] xen/netfront: unconditionally initialize smartpoll hrtimer

Make sure smartpoll.timer is properly initialized, even if we're not
using it, so that hrtimer operations on it still work OK.  Specifically,
this fixes a crash when suspending, as it tries to cancel any outstanding
timer.

Signed-off-by: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
---
 drivers/net/xen-netfront.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/xen-netfront.c b/drivers/net/xen-netfront.c
index 6661d6d..47e6a71 100644
--- a/drivers/net/xen-netfront.c
+++ b/drivers/net/xen-netfront.c
@@ -1654,9 +1654,9 @@ static int xennet_connect(struct net_device *dev)
 			np->smart_poll.feature_smart_poll = 0;
 	}
 
+	hrtimer_init(&np->smart_poll.timer, CLOCK_MONOTONIC,
+		     HRTIMER_MODE_REL);
 	if (np->smart_poll.feature_smart_poll) {
-		hrtimer_init(&np->smart_poll.timer, CLOCK_MONOTONIC,
-			     HRTIMER_MODE_REL);
 		np->smart_poll.timer.function = smart_poll_function;
 		np->smart_poll.netdev = dev;
 		np->smart_poll.smart_poll_freq = DEFAULT_SMART_POLL_FREQ;
-- 
1.7.1

